enhancements/installer/aws-custom-edge-machineset-local-zones.md

```diff
@@ -0,0 +1,952 @@
+when installing in existing VPC.
+
+The subnet will be classified to be used on the `edge` machine pool on AWS installation
+when the [zoneType](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_AvailabilityZone.html) attribute for the `AvailabilityZone` has the value of `local-zone`.
+The regular zones, `zoneType==availability-zone`, will keep the same flow used on the `worker`
+and control plane pool: installer lookup all the regular zones on the list of subnet-IDs
+provided on `platform.aws.subnets`.
+
+Currently, Local Zones does not support network load balancers (NLB or CLB) and Nat Gateways.
+Due to the network limitations of AWS Local Zones, the installer requires that the subnet created on
+Local Zones must be attached to the public route table.
+
+#### instance type
+
+Currently, the logic to choose the instance type will check the preferred instance type
+across the zones available to create the nodes (control plane and workers), querying the
+API [DescribeInstanceTypeOfferings](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstanceTypeOfferings.html).
+
+The [instance type offerings](https://aws.amazon.com/ec2/pricing/on-demand/) on
+Local Zones are different than the regular zones (see [exported instance offerings matrix on US Local Zones](https://gist.github.com/mtulio/c98aa15128a7becb06a372f00d824c42)).
+For that reason, the installer will lookup the offering for each local zone along
+the [preferred types](https://github.com/openshift/installer/blob/master/pkg/types/aws/defaults/platform.go#L38) when
+creating the MachineSets manifests for the `edge` pools. To cover all Local Zones
+location offerings, the type `c5d.2xlarge` will be appended to the preferred list.
+With that strategy, all the current GA locations will be covered with supported instance
+types, and when the most recent instances (currently 6th-gen `m6i.xlarge`) will be
+offered on those locations, it will be selected automatically without needing to
+change the installer.
+
+The logic of the preferred instance type will be replaced for the `edge` pool when
+the user provides the instance type on the install-config, attribute `compute[name=edge].platform.aws.type`.
+
+#### Volume type
+
+The EBS Volume Type `gp3`, used by default on IPI, is not broadly available
+across all GA Local Zones locations. [The Local Zones Features page](https://aws.amazon.com/about-aws/global-infrastructure/localzones/features) is the current place to check which location supports `gp3`.
+
+AWS also does not provide an API to query the zone offering for EBS flavors. The EBS
+volume type for Local Zones, `edge` machine pool, will be fixed to `gp2`.
+
+The EBS Volume Type can be changed on the `edge` machine pool option
+`compute[name=edge].platform.aws.rootVolume.type` on the `install-config.yaml`.
+
+If the user-workloads requires an Phisical Volume, the `storageClass` can be set to `gp2-csi`
+on the deployment config.
```

Comment:
What's "Phisical Volume"? Do you mean PersistentVolume Kubernetes object?

Reply:
Yes, typo. Thanks for getting that. I will fix it asap.
We are working to add this[1] to the documentation to help users and inform them about the limitations of [2]. Open to suggestions on that user-facing doc too. Thanks

[1] https://github.com/openshift/openshift-docs/pull/57063
[2] https://github.com/openshift/openshift-docs/pull/57736

Reply:
Documentation is live on 4.12+ on the Cluster tasks exemplifying workloads deployment in Local Zone nodes, including PV volumes:
https://docs.openshift.com/container-platform/4.12/post_installation_configuration/cluster-tasks.html#installation-extend-edge-nodes-aws-local-zones_post-install-cluster-tasks